### PR TITLE
[MRG] fix loading without age information / single-column .tsv

### PR DESCRIPTION
**Pull request by skjerns**
PR Description
--------------

When loading participants.tsv with only a single column (only participant ids), `p_age` is empty, and can therefore not be created. See also #891 . This simple fix resolves this problem.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Review by sappelhoff — COMMENTED** [review]
Thanks for the fix @skjerns - would you mind adding a small test here? --> https://github.com/mne-tools/mne-bids/blob/main/mne_bids/tests/test_report.py

also: it's just a small fix, but could you add an entry to the changelog? https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst

It'll also showcase your first PR contribution to mne-bids :-)

File: mne_bids/report.py [hunk 1]
```diff
@@ -229,6 +229,7 @@ def _summarize_participants_tsv(root):
     p_ages = participants_tsv.get('age')
     min_age, max_age = 'n/a', 'n/a'
     mean_age, std_age = 'n/a', 'n/a'
+    print(p_ages,participants_tsv)
```
Comment (hoechenberger) [inline mne_bids/report.py:232]: This should be removed

File: mne_bids/tests/test_report.py [hunk 2]
```diff
@@ -59,3 +60,40 @@ def test_report(tmp_path):
...
 
     assert report == expected_report
+
+
+@pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
+def test_report_no_participant_information(tmp_path):
+    """
+    Test that report generated works as intended even when all participant
+    information such as age, gender, handedness is removed from the tsv file.
+    """
```
Comment (hoechenberger) [inline mne_bids/tests/test_report.py:70]:
```suggestion
    """Test report with participants.tsv with participant_id column only."""
```

File: mne_bids/tests/test_report.py [hunk 3]
```diff
@@ -59,3 +60,40 @@ def test_report(tmp_path):
...
+    bids_root = str(tmp_path)
```
Comment (hoechenberger) [inline mne_bids/tests/test_report.py:71]: No need to convert to a string
```suggestion
    bids_root = tmp_path
```

File: mne_bids/tests/test_report.py [hunk 4]
```diff
@@ -59,3 +60,40 @@ def test_report(tmp_path):
...
+    raw = mne.io.read_raw_fif(raw_fname, verbose=False)
+    raw.info['line_freq'] = 60
+    bids_path.update(root=bids_root)
```
Comment (hoechenberger) [inline mne_bids/tests/test_report.py:74]: Avoid altering the original
```suggestion
    bids_path = bids_path.copy().update(root=bids_root)
```

File: mne_bids/tests/test_report.py [hunk 5]
```diff
@@ -59,3 +60,40 @@ def test_report(tmp_path):
...
+    write_raw_bids(raw, bids_path, overwrite=True, verbose=False)
+    
+    # remove all information and check if report still runs
+    os.remove(op.join(bids_root, 'participants.json'))
```
Comment (hoechenberger) [inline mne_bids/tests/test_report.py:78]:
```suggestion
    (bids_root / 'participants.json').unlink()
```

File: mne_bids/tests/test_report.py [hunk 6]
```diff
@@ -59,3 +60,40 @@ def test_report(tmp_path):
...
+    
+    # overwrite participant information to see if report still runs
+    with open(op.join(bids_root, 'participants.tsv'), 'w') as f:
+        f.write('participant_id\nsub-001')
```
Comment (hoechenberger) [inline mne_bids/tests/test_report.py:82]: Simpler
```suggestion
    (bids_root / 'participants.tsv').write_text('participant_id\nsub-001')
```

File: doc/whats_new.rst [hunk 7]
```diff
@@ -83,6 +84,8 @@ Requirements
 Bug fixes
 ^^^^^^^^^
 
+- Fix setting unknown ages if no age parameter is supplied in csv in function :func:`mne_bids.report._summarize_participants_tsv` (:gh:`912`)
```
Comment (hoechenberger) [inline doc/whats_new.rst]: This is documenting a change in a private function, so this is not directly user-visible and shouldn't go into the changelog. Better do something like:

```suggestion
- :func:`mne_bids. make_report` can now handle `participants.tsv` files without an `age` column, by `Simon Kern`_ (:gh:`912`)
```

File: mne_bids/tests/test_report.py (hunk 3 above)
Comment (skjerns) [reply mne_bids/tests/test_report.py:71]: sure, just copied it from the other test

**Review by sappelhoff — DISMISSED** [review]
something seems to have gone wrong with rebasing and/or merging `main` into this PR :thinking: There are lots of changes that shouldn't be changes

**Comment by skjerns** [comment]
> something seems to have gone wrong with rebasing and/or merging `main` into this PR 🤔 There are lots of changes that shouldn't be changes

damn, I might have pulled from the wrong main. I'll tried to fix it - is it fine now?

**Review by sappelhoff — COMMENTED** [review]
looks much better now!

File: doc/whats_new.rst [hunk 8]
```diff
@@ -54,6 +54,8 @@ Bug fixes
 
 - ...
 
+- Fix setting unknown ages if no age parameter is supplied in csv in function :func:`mne_bids.make_report` (:gh:`912`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:57]: Should be rephrased, I don't think it's very clear what is meant here. I can give it a shot later, currently on my phone

**Comment by hoechenberger** [comment]
Whoops, CI failing now, can you have a look? If you're busy, I might be able to take over tomorrow

File: doc/whats_new.rst (hunk 8 above)
Comment (skjerns) [reply doc/whats_new.rst]: I tried (also to re-trigger CI), better now?

Comment (skjerns) [reply doc/whats_new.rst]: at least CI seems to work now :)

Comment (skjerns) [reply doc/whats_new.rst]: `Fix loading tsv participant summary information when only subject_id is supplied and all other information (age, gender, handedness) is missing in :func:`mne_bids.make_report` (:gh:`912`)`

File: doc/whats_new.rst [hunk 9]
```diff
@@ -54,6 +54,8 @@ Bug fixes
 
 - Forcing EDF conversion in :func:`mne_bids.write_raw_bids` properly uses the ``overwrite`` parameter now, by `Adam Li`_ (:gh:`930`)
 
+- Fix loading tsv participant summary information when only subject_id is supplied and all other information (age, gender, handedness) is missing in :func:`mne_bids.make_report` (:gh:`912`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:57]:
```suggestion
- :func:`mne_bids.make_report` now correctly handles `participant.tsv` files that only contain a `subject_id` column, by `Simon Kern`_ (:gh:`912`)
```

File: mne_bids/tests/test_report.py [hunk 10]
```diff
@@ -33,6 +33,7 @@
     channel_unit_changed='ignore:The unit for chann*.:RuntimeWarning:mne',
 )
 
+bids_path2 = bids_path.copy()
```
Comment (hoechenberger) [inline mne_bids/tests/test_report.py:36]: this shouldn't be necessary, I'll have a look at this right now

File: doc/whats_new.rst (hunk 8 above)
Comment (hoechenberger) [reply doc/whats_new.rst]: I pushed another small adjustment to make it more concise

**Review by hoechenberger — APPROVED** [review]

**Review by sappelhoff — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @skjerns

**Pull request merged**